src/uncompress.rs

```diff
@@ -48,33 +60,43 @@ impl Uncompress {
             ))
             .finally()?
             .wait_with_output()?;
-        ensure!(
-            out.status.success(),
-            format!("Extract process exited with error:{:#?}", out)
-        );
-        Ok(())
+
+        if !out.status.success() {
+            Err(Error::CommandError(
+                String::from_utf8_lossy(&out.stderr).to_string(),
+            ))
+        } else {
+            Ok(())
+        }
     }
 
-    pub fn unzip(self, dir: &Path) -> Result<(), failure::Error> {
+    pub fn unzip(self, dir: &Path) -> Result<()> {
         let out = self
             .0
             .then(&format!("unzip -X -d {}", dir.display()))
             .finally()?
             .wait_with_output()?;
-        ensure!(
-            out.status.success(),
-            format!("Extract process exited with error:{:#?}", out)
-        );
-        Ok(())
+
+        if !out.status.success() {
+            Err(Error::CommandError(
+                String::from_utf8_lossy(&out.stderr).to_string(),
+            ))
+        } else {
+            Ok(())
+        }
     }
 
-    pub fn file(self, target: &Path) -> Result<(), failure::Error> {
+    pub fn file(self, target: &Path) -> Result<()> {
         let out = self.0.finally()?.wait_with_output()?;
-        ensure!(
-            out.status.success(),
-            format!("Extract process exited with error:{:#?}", out)
-        );
+
         fs::File::create(target)?.write_all(&out.stdout)?;
-        Ok(())
+
+        if !out.status.success() {
+            Err(Error::CommandError(
```

Comment:
This `Error::CommandError` constructor is used a feel times using the same logic to be generated from a `std::process::Output`, I think it would be nice if we had a `impl From<std::process::Output> for Error`, so we avoid boilerplate here and we can make the code more maintainable.